tests/build/multi-platform.go

```diff
@@ -411,8 +421,617 @@ var _ = framework.MultiPlatformBuildSuiteDescribe("Multi Platform Controller E2E
+			})
+
+			It("check cleanup happened successfully", func() {
+				Eventually(func() error {
+					instances, err := getIbmZDynamicInstances(dynamicInstanceTag)
+					if err != nil {
+						return err
+					}
+					if len(instances) != 0 {
+						return fmt.Errorf("instance is not cleaned up properly, current running instances: %v", instances)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed(), "timed out when verifying that the remote host was cleaned up correctly")
+			})
+
+		})
+	})
+	Describe("ibmp dynamic allocation", Label("ibmp-dynamic"), func() {
+		var testNamespace, applicationName, componentName, multiPlatformSecretName, multiPlatformTaskName, dynamicInstanceTag, instanceId string
+		var component *appservice.Component
+		var timeout, interval time.Duration
+
+		AfterAll(func() {
+			// Cleanup ibm key & ssh secrets and host-config
+			Expect(f.AsKubeAdmin.CommonController.DeleteSecret(ControllerNamespace, IbmSecretName)).To(Succeed())
+			Expect(f.AsKubeAdmin.CommonController.DeleteSecret(ControllerNamespace, SshSecretName)).To(Succeed())
+			Expect(f.AsKubeAdmin.CommonController.DeleteConfigMap(HostConfig, ControllerNamespace, true)).To(Succeed())
+
+			//Forcefully remove instance incase it is not removed by multi-platform-controller
+			err = terminateIbmPInstance(instanceId)
+			if err != nil {
+				GinkgoWriter.Printf("error terminating instance again: %v", err)
+			}
+
+			if !CurrentSpecReport().Failed() {
+				Expect(f.AsKubeAdmin.HasController.DeleteComponent(componentName, testNamespace, false)).To(Succeed())
+				Expect(f.AsKubeAdmin.HasController.DeleteApplication(applicationName, testNamespace, false)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			} else {
+				Expect(f.AsKubeAdmin.CommonController.StoreAllPods(testNamespace)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.StoreAllPipelineRuns(testNamespace)).To(Succeed())
+			}
+		})
+
+		BeforeAll(func() {
+
+			f, err = framework.NewFramework(utils.GetGeneratedNamespace("multi-platform-ibmp"))
+			Expect(err).NotTo(HaveOccurred())
+			testNamespace = f.UserNamespace
+			Expect(testNamespace).NotTo(BeNil(), "failed to create sandbox user namespace")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Restart multi-platform-controller pod to reload configMap again
+			podList, err := f.AsKubeAdmin.CommonController.ListAllPods(ControllerNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			for i := range podList.Items {
+				podName := podList.Items[i].Name
+				if strings.HasPrefix(podName, ControllerNamespace) {
+					err := f.AsKubeAdmin.CommonController.DeletePod(podName, ControllerNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+				}
+			}
+
+			//sleep for 10 sec so that multi-platform-controller pod comes up
+			time.Sleep(10 * time.Second)
```

Comment:
Any reason not use `Eventually` to wait for the pod comes up?

Reply:
makes sense.